src/main/scala/tech/cryptonomic/conseil/Lorre.scala

```diff
@@ -1,729 +1,167 @@
-   * @param events the relevant levels, each with its own selection pattern, that calls for a refresh
-   */
-  private def processAccountRefreshes(events: AccountUpdatesEvents): Future[AccountUpdatesEvents] =
-    if (events.nonEmpty) {
-      for {
-        storedHead <- apiOperations.fetchMaxLevel
-        updated <- if (events.exists(_._1 <= storedHead)) {
-          val (past, toCome) = events.partition(_._1 <= storedHead)
-          val (levels, selectors) = past.unzip
-          logger.info(
-            "A block was reached that requires an update of account data as specified in the configuration file. A full refresh is now underway. Relevant block levels: {}",
-            levels.mkString(", ")
-          )
-          apiOperations.fetchBlockAtLevel(levels.max).flatMap {
-            case Some(referenceBlockForRefresh) =>
-              val (hashRef, levelRef, timestamp, cycle) =
-                (
-                  BlockHash(referenceBlockForRefresh.hash),
-                  referenceBlockForRefresh.level,
-                  referenceBlockForRefresh.timestamp.toInstant(),
-                  referenceBlockForRefresh.metaCycle
-                )
-              db.run(
-                  //put all accounts in checkpoint, log the past levels to the db, keep the rest for future cycles
-                  TezosDb.refillAccountsCheckpointFromExisting(hashRef, levelRef, timestamp, cycle, selectors.toSet) >>
-                      TezosDb.writeProcessedEventsLevels(
-                        ChainEvent.accountsRefresh.render,
-                        levels.map(BigDecimal(_)).toList
-                      )
-                )
-                .andThen {
-                  case Success(accountsCount) =>
-                    logger.info(
-                      "Checkpoint stored for{} account updates in view of the full refresh.",
-                      accountsCount.fold("")(" " + _)
-                    )
-                  case Failure(err) =>
-                    logger.error(
-                      "I failed to store the accounts refresh updates in the checkpoint",
-                      err
-                    )
-                }
-                .map(_ => toCome) //keep the yet unreached levels and pass them on
-            case None =>
-              logger.warn(
-                "I couldn't find in Conseil the block data at level {}, required for the general accounts update, and this is actually unexpected. I'll retry the whole operation at next cycle.",
-                levels.max
-              )
-              Future.successful(events)
-          }
-        } else Future.successful(events)
-      } yield updated
-    } else Future.successful(events)
-
-  /**
-    * Fetches all blocks not in the database from the Tezos network and adds them to the database.
-    * Additionally stores account references that needs updating, too
+  /** This is the core logic of Lorre, based on additional resources and services
```

Comment:
I would prefer to see everything from this line below in the separate file.

Reply:
I'll check this out more throughly then.